Mod Menu integration

Leo40Git opened the pull request:
*Alternate title: "Avenging #18 and #29"*
- Added "Client" badge to mod listing in the Mod Menu
- Added config button for mod listing that sends you to Sodium's video settings GUI

Leo40Git committed:
Add "Client" badge to Mod Menu listing

I'm guessing my PR was ignored the last time because I didn't notice a "custom" object already existed and made another one :V

Leo40Git committed:
Add Mod Menu integration: config button that sends you to the Sodium options GUI

Leo40Git commented:
Also, while we're talking entrypoints, may I ask why we have a main entrypoint that does literally nothing? (me.jellysquid.mods.sodium.common.SodiumMod)
Should probably be removed, and the "environment" entry in the mod JSON should be "client". Sodium is entirely clientside rendering, after all.

jellysquid3 commented:
> may I ask why we have a main entrypoint that does literally nothing

It's a leftover from my usual project template. It's not serving any purpose and can be removed.

Leo40Git committed:
Make mod actually clientside ;)

CDAGaming commented on build.gradle:
```diff
@@ -33,6 +33,10 @@ dependencies {
     minecraft "com.mojang:minecraft:${project.minecraft_version}"
     mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
     modCompile "net.fabricmc:fabric-loader:${project.loader_version}"
```
While technically out of scope, perhaps this could become modImplementation instead of modCompile

CDAGaming commented on src/main/resources/fabric.mod.json:
```diff
@@ -13,17 +13,18 @@
   },
```
While also out of scope, this pr could be improved by adding an issues argument to the contact section above, allowing the Issues button to be available in the Mod Menu as well

CDAGaming reviewed (approved):
These are just some suggestions that could make this better, though in my experience it worked as intended

Edit: LGTM after reviewed commits, no further suggestions, nice job!

Leo40Git committed:
Review changes

- change deprecated modCompile to modImplementation
- add issues link to contacts so Mod Menu's "Issues" button works

Leo40Git committed:
Update Loom to 0.4

Leo40Git commented:
Closed in favor of #114.
